lib/repl.js

```diff
@@ -149,7 +155,7 @@ function REPLServer(prompt,
   self.useGlobal = !!useGlobal;
   self.ignoreUndefined = !!ignoreUndefined;
   self.replMode = replMode || exports.REPL_MODE_SLOPPY;
-  self.underscoreAssigned = false;
+  self.underscoreAssigned = self.noUnderscore = !!noUnderscore;
```

Comment:
In this line 😃 it's a double assignment, not a comparison. (It's late here, I had too loke twice too).
```js
self.underscoreAssigned = (self.noUnderscore = !!noUnderscore);
```